### PR TITLE
explicitly set container interface mac to local unicast

### PR DESCRIPTION
**Pull request by phoracek**
With prefix 02:00:00:, we get local unicast MAC address from non-reserved range.

Addresses https://github.com/kubevirt/kubevirt/issues/1494

### CODE REVIEW

**Comment by phoracek** [comment]
Passes plugin test and assigns MAC with 02:00:00 prefix to container side of veth pair.

@SchSeba @yuvalif could you please review?

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -98,8 +110,20 @@ func setupVeth(contNetns ns.NetNS, contIfaceName string) (*current.Interface, *c
...
 			return err
 		}
+
+		containerLink, err := netlink.LinkByName(containerVeth.Name)
+		if err != nil {
+			return fmt.Errorf("failed to lookup %q: %v", containerVeth.Name, err)
+		}
+
+		containerMac := generateRandomMac()
+		err = netlink.LinkSetHardwareAddr(containerLink, containerMac)
```
Comment (yuvalif) [inline pkg/plugin/plugin.go]: will this prevent duplicates?

Comment (phoracek) [reply pkg/plugin/plugin.go]: It won't, but there are not many interfaces in the pod, do you think there is real risk of collisions?

Comment (phoracek) [reply pkg/plugin/plugin.go]: But with only 3 octets used, I guess it could happen. Fixing

**Comment by phoracek** [comment]
@yuvalif added dirty fix for the collision, I feel bad (just a little).

**Comment by phoracek** [comment]
@yuvalif sorry, I'm just bad at maths, there are pretty low chances of a collision.

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -98,8 +110,25 @@ func setupVeth(contNetns ns.NetNS, contIfaceName string) (*current.Interface, *c
...
 			return err
 		}
+
+		containerLink, err := netlink.LinkByName(containerVeth.Name)
+		if err != nil {
+			return fmt.Errorf("failed to lookup %q: %v", containerVeth.Name, err)
+		}
+
+		// In case the MAC address is already assigned to another interface, retry
+		macSetupRetries := 2
```
Comment (SchSeba) [inline pkg/plugin/plugin.go]: Can you configure this as a const?

**Comment by phoracek** [comment]
@SchSeba done

**Pull request merged**